src/ansys/simai/core/api/mixin.py

```diff
@@ -154,6 +156,11 @@ def download_file(
             None if a file is provided or a ``BytesIO`` object with the file's
             content otherwise.
         """
+        if dir is not None and file is None:
```

Comment:
We could just `if file.is_dir():`, it would be less weird to use than having a `dir` arg that only works when `file` is `None`.

Reply:
To me, this makes more sense to have a dedicate argument for dir, for having it explicit.